hknweb/academics/models/course_surveys.py

```diff
@@ -0,0 +1,29 @@
+from django.db import models
+
+from .base_models import AcademicEntity
+
+
+class Question(AcademicEntity):
+    pass
+
+
+class Survey(AcademicEntity):
+    # reference attributes
+    survey_icsr = models.ForeignKey('ICSR', on_delete=models.CASCADE, related_name='survey_icsr')
+
+    # value attributes
+    num_students = models.IntegerField()
+    response_count = models.IntegerField()

```

Comment:
Change to [PositiveIntegerField](https://docs.djangoproject.com/en/3.0/ref/models/fields/#positiveintegerfield) to disallow negative students / responses?

Reply:
Both fields should be at least nonnegative. I can't see a scenario where `num_students` will ever be 0, so that will use a `PositiveIntegerField`. However, we have several cases where `response_count` is 0, so the plan is to provide validation so that `response_count` is only ever non-negative.